types/agent-base/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for agent-base 4.2
+// Project: https://github.com/TooTallNate/node-agent-base#readme
+// Definitions by: Christopher Quadflieg <https://github.com/Shinigami92>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+import { EventEmitter } from 'events';
+
+declare namespace Agent {
+	type AgentCallback = (
+		req?: any,
+		opts?: {
+			secureEndpoint: boolean;
+		}
+	) => void;
+
+	interface AgentOptions {
+		timeout?: number;
+		host?: string;
+		port?: number;
+		[key: string]: any;
+	}
+
+	interface Agent extends EventEmitter {
+		_promisifiedCallback: boolean;
+		timeout: number | null;
+		options?: AgentOptions;
+		callback: AgentCallback;
+		addRequest: (req?: any, opts?: any) => void;
+		freeSocket: (socket: any, opts: any) => void;
+	}
+}
+
+/**
+ * Base `http.Agent` implementation.
+ * No pooling/keep-alive is implemented by default.
+ */
+declare function Agent(opts?: Agent.AgentOptions): Agent.Agent;
+declare function Agent(
+	callback: Agent.AgentCallback,
```

Comment:
Call back needs to be typed as per https://github.com/TooTallNate/node-agent-base#callbackhttpclientrequest-req-object-options-function-cb--undefined

Reply:
I can not find any use of a third parameter cb, it is never used or called?!
I think the documentation is not very representative :/

Reply:
And for the `AgentCallback#(req: any, ...)` I can add at most a few methods and properties that are used in the code
But I am not sure if that is effective